src/registries/npm-registry.js

```diff
@@ -50,16 +52,14 @@ export default class NpmRegistry extends Registry {
     return name.replace('/', '%2f');
   }
 
-  request(pathname: string, opts?: RegistryRequestOptions = {}): Promise<*> {
-    const registry = addSuffix(this.getRegistry(pathname), '/');
+  request(pathname: string, opts?: RegistryRequestOptions = {}, packageName: ?string): Promise<*> {
+    const registry = this.getRegistry(packageName || pathname);
     const requestUrl = url.resolve(registry, pathname);
-    const alwaysAuth = this.getScopedOption(registry.replace(/^https?:/, ''), 'always-auth')
-      || this.getOption('always-auth')
-      || removePrefix(requestUrl, registry)[0] === '@';
+    const alwaysAuth = this.getRegistryOrGlobalOption(registry, 'always-auth');
```

Comment:
@bestander does it make sense to force adding auth key to resoled url?
I spent a day some months ago when tried to fix [similar problem](https://github.com/yarnpkg/yarn/issues/521#issuecomment-277500043) with our artifactory.